actionview/lib/action_view/helpers/text_helper.rb

```diff
@@ -467,18 +467,19 @@ def cut_excerpt_part(part_position, part, separator, options)
           radius   = options.fetch(:radius, 100)
           omission = options.fetch(:omission, "...")
 
-          part = part.split(separator)
-          part.delete("")
-          affix = part.size > radius ? omission : ""
+          if separator != ""
+            part = part.split(separator)
+            part.delete("")
+          end
 
-          part = if part_position == :first
-            drop_index = [part.length - radius, 0].max
-            part.drop(drop_index)
-          else
-            part.first(radius)
+          affix = part.length > radius ? omission : ""
+          part = part.public_send(part_position == :first ? :last : :first, radius)
```

Comment:
This is complex enough that I prefer the longer version from your benchmark script:

```suggestion

          part =
            if part_position == :first
              part.last(radius)
            else
              part.first(radius)
            end
```

Reply:
We'll have to require `active_support/core_ext/string/access` in order to use `String#first` and `String#last` here.